### PR TITLE
Model Registry: Initial content of model details tab with stub editable fields and tab navigation handling

### PR DESCRIPTION
**Pull request by mturley**
Part of [RHOAIENG-2235](https://issues.redhat.com/browse/RHOAIENG-2235)

## Description

Implements most of the "Details" tab of the registered model page. Note that the "Properties" section is not functional and will be implemented in a followup PR. Also note that while the Description and Labels sections have editable functionality implemented, the actual last step of persisting the edits to the API is a stub promise that simply waits 2 seconds in order to show the "saving" state (edits are not allowed while saving). Implementing the real API patch requests will be in a separate PR.

Model details tab:
Screenshot 2024-04-29 at 7 31 32 PM

Description field in edit mode:
Screenshot 2024-04-29 at 7 15 02 PM

Description field disabled while saving (currently a stub 2-second-delay promise):
Screenshot 2024-04-29 at 7 24 38 PM

Labels in edit mode:
Screenshot 2024-04-29 at 7 15 13 PM

Editing an existing label:
Screenshot 2024-04-29 at 7 15 21 PM

Add label modal:
Screenshot 2024-04-29 at 7 24 49 PM
Screenshot 2024-04-29 at 7 24 56 PM

Label editing disabled while saving (currently a stub 2-second-delay promise):
Screenshot 2024-04-29 at 7 25 53 PM

## How Has This Been Tested?

1. Go to `Model Registry` tab
2. Click on any registered model in the `Registered Models` table and then click the `Details` tab, or click the kebab menu at the right of a registered model and click `View details`
3. You will be taken to the model details view shown above
4. Click between the tabs and note that the URL route is changing and using the new `/details` and `/versions` subpaths

## Test Impact

Will be added shortly.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/modelRegistry/ModelRegistryRoutes.tsx
Comment (lucferbux) [inline frontend/src/pages/modelRegistry/ModelRegistryRoutes.tsx:14]: Ok, this might change with my recent changes, can be simplified

File: frontend/src/pages/modelRegistry/screens/ModelDetailsView.tsx
Comment (lucferbux) [inline frontend/src/pages/modelRegistry/screens/ModelDetailsView.tsx:18]: I would assume we don't wanna console log this at the end right?

File: frontend/src/pages/modelRegistry/screens/useRegisteredModelUrl.ts
Comment (lucferbux) [inline frontend/src/pages/modelRegistry/screens/useRegisteredModelUrl.ts:5]: Yeah, good call to have a utils for that, we might refactor it into another place but it's a great strat.

**Review by lucferbux — CHANGES_REQUESTED** [review]
Ok, so far it's working great, this is going really great 👌

File: frontend/src/pages/modelRegistry/screens/ModelDetailsView.tsx
Comment (mturley) [reply frontend/src/pages/modelRegistry/screens/ModelDetailsView.tsx:18]: True, I can remove that. Although without any code outside the JSX here the linter wants this to be an inline arrow-function expression, which means when we later consume the context to do API calls that diff will change a bunch of whitespace in here and cause conflicts if multiple people are working on this file. Probably not an issue, will remove.

File: frontend/src/pages/modelRegistry/ModelRegistryRoutes.tsx
Comment (mturley) [reply frontend/src/pages/modelRegistry/ModelRegistryRoutes.tsx:14]: Ok, once yours is merged I can rebase. How do you suggest we simplify? do you just mean the `path` here will be your simpler `registeredModels/:registeredModelId` and this will otherwise be the same (we want this new nesting)?

File: frontend/src/components/EditableLabelsDescriptionListGroup.tsx
Comment (lucferbux) [inline frontend/src/components/EditableLabelsDescriptionListGroup.tsx:36]: I feel that if this is gonna be reused in the details view we can move it into a utils right?
Not needed right now, can be refactor on @dpanshug pr

**Review by lucferbux — APPROVED** [review]
/lgtm

File: frontend/src/components/EditableLabelsDescriptionListGroup.tsx
Comment (lucferbux) [reply frontend/src/components/EditableLabelsDescriptionListGroup.tsx:36]: Resolving conversation it can be done in another PR

**Comment by lucferbux** [comment]
Ok, so far it's going great! This might unblock the rest of the work!

**Pull request merged**
